pkg/oc/cli/admin/node/logs/logs.go

```diff
@@ -0,0 +1,489 @@
+			node: info.Name,
+			req:  req,
+			raw:  o.Raw || o.Path == "journal",
+		})
+		return nil
+	})
+	if err != nil {
+		if agg, ok := err.(errors.Aggregate); ok {
+			for _, err := range agg.Errors() {
+				errs = append(errs, err)
+			}
+		} else {
+			errs = append(errs, err)
+		}
+	}
+
+	found := len(errs) + len(requests)
+	skipPrefix := found == 1 && result.TargetsSingleItems()
+	unify := o.Path == "journal" || o.Unify
+	out := bufio.NewWriterSize(o.Out, 1024*16)
+	defer out.Flush()
+
+	if unify {
+		var readers []Reader
+		for i := range requests {
+			req := requests[i]
+			req.skipPrefix = true
+			pr, pw := io.Pipe()
+			readers = append(readers, Reader{
+				R: pr,
+			})
+			go func() {
+				err := req.WriteTo(pw)
+				pw.CloseWithError(err)
+			}()
+		}
+		_, err := NewMergeReader(readers...).WriteTo(out)
+		if agg := errors.Flatten(errors.NewAggregate([]error{err})); agg != nil {
+			errs = append(errs, agg.Errors()...)
+		}
+
+	} else {
+		for _, req := range requests {
+			req.skipPrefix = skipPrefix
+			if err := req.WriteTo(out); err != nil {
+				errs = append(errs, err)
+			}
+		}
+	}
+
+	if len(errs) > 0 {
+		for _, err := range errs {
+			fmt.Fprintf(o.ErrOut, "error: %v\n", err)
+		}
+		return kcmdutil.ErrExit
+	}
+
+	return nil
+}
+
+func optionallyDecompress(out io.Writer, in io.Reader) error {
+	bufferSize := 4096
+	buf := bufio.NewReaderSize(in, bufferSize)
+	head, err := buf.Peek(1024)
+	if err != nil && err != io.EOF {
+		return err
+	}
+	if _, err := gzip.NewReader(bytes.NewBuffer(head)); err != nil {
+		// not a gzipped stream
+		_, err = io.Copy(out, buf)
+		return err
+	}
+	r, err := gzip.NewReader(buf)
+	if err != nil {
+		return err
+	}
+	_, err = io.Copy(out, r)
+	return err
+}
+
+func outputDirectoryEntriesOrContent(out io.Writer, in io.Reader, prefix []byte) error {
+	bufferSize := 4096
+	buf := bufio.NewReaderSize(in, bufferSize)
+
+	// turn href links into lines of output
+	content, _ := buf.Peek(bufferSize)
+	if bytes.HasPrefix(content, []byte("<pre>")) {
+		reLink := regexp.MustCompile(`href="([^"]+)"`)
+		s := bufio.NewScanner(buf)
+		s.Split(func(data []byte, atEOF bool) (advance int, token []byte, err error) {
+			matches := reLink.FindSubmatchIndex(data)
+			if matches == nil {
+				advance = bytes.LastIndex(data, []byte("\n"))
+				if advance == -1 {
+					advance = 0
+				}
+				return advance, nil, nil
+			}
+			advance = matches[1]
+			token = data[matches[2]:matches[3]]
+			return advance, token, nil
+		})
+		for s.Scan() {
+			if _, err := out.Write(prefix); err != nil {
+				return err
+			}
+			if _, err := fmt.Fprintln(out, s.Text()); err != nil {
+				return err
+			}
+		}
+		return s.Err()
+	}
+
+	// without a prefix we can copy directly
+	if len(prefix) == 0 {
+		_, err := io.Copy(out, buf)
+		return err
+	}
+
+	r := NewMergeReader(Reader{R: buf, Prefix: prefix})
+	_, err := r.WriteTo(out)
+	return err
+}
+
+type Reader struct {
+	R      io.Reader
```

Comment:
When using public member please make the name meaningful, or just embed `io.Reader`.

Reply:
Was just being consistent with how the `io` package names structs like this.